src/setfit/trainer.py

```diff
@@ -186,15 +195,24 @@ def train(self):
     def evaluate(self):
         """Computes the metrics for a given classifier."""
         self._validate_column_mapping(self.eval_dataset)
+
         if self.column_mapping is not None:
             logger.info("Applying column mapping to evaluation dataset")
             self.eval_dataset = self._apply_column_mapping(self.eval_dataset, self.column_mapping)
-        metric_fn = evaluate.load(self.metric)
+
+        metric_config = "multilabel" if self.model.multi_target_strategy is not None else None
+        metric_fn = evaluate.load(self.metric, config_name=metric_config)
+
         x_test = self.eval_dataset["text"]
-        y_test = self.eval_dataset["label"]
+
+        if self.model.multi_target_strategy:
```

Comment:
Similar comment as above - the column mapping logic in #75 should ensure that the `eval_dataset` always has a `text` and `label` column, which means we can remove this if/else logic